rskj-core/src/test/java/co/rsk/peg/BridgeTest.java

```diff
@@ -2266,6 +2266,40 @@ void getProposedFederationAddress(MessageCall.MsgType msgType, ActivationConfig
         }
     }
 
+    @ParameterizedTest()
+    @MethodSource("msgTypesAndActivations")
+    void getProposedFederationCreationBlockNumber(MessageCall.MsgType msgType, ActivationConfig activationConfig) throws VMException {
+        Transaction rskTxMock = mock(Transaction.class);
+        doReturn(true).when(rskTxMock).isLocalCallTransaction();
+
+        BridgeSupport bridgeSupportMock = mock(BridgeSupport.class);
+        long expectedCreationBlockNumber = 123456L;
+        when(bridgeSupportMock.getProposedFederationCreationBlockNumber()).thenReturn(Optional.of(expectedCreationBlockNumber));
+
+        Bridge bridge = bridgeBuilder
+            .transaction(rskTxMock)
+            .activationConfig(activationConfig)
+            .bridgeSupport(bridgeSupportMock)
+            .msgType(msgType)
+            .build();
+
+        CallTransaction.Function function = BridgeMethods.GET_PROPOSED_FEDERATION_CREATION_BLOCK_NUMBER.getFunction();
+        byte[] data = function.encode();
+
+        if (activationConfig.isActive(ConsensusRule.RSKIP419, 0)) {
```

Comment:
Same here. I would instead go with multiple-test methods, eliminating the need for the conditional. To me, the test would be easier to follow in that way.

Reply:
I agree.